core/src/main/java/io/grpc/internal/DelayedClientTransport2.java

```diff
@@ -85,13 +87,50 @@
   @Nullable
   private SubchannelPicker lastPicker;
 
-  DelayedClientTransport2(Executor streamCreationExecutor) {
-    this.streamCreationExecutor = streamCreationExecutor;
+  @GuardedBy("lock")
+  private long lastPickerVersion;
```

Comment:
In case of pick-first, there is only one Subchannel, thus the picker may return it in `IDLE`, `CONNECTING` and `READY`. `CONNECTING->READY` must trigger `updatePicker` to get the delayed streams reprocessed. Without this change, the LoadBalancer *must* create a new picker instance, even if it's literally the same, otherwise new streams that race with `updatePicker` may have this failure mode:
1. T1: `DelayedClientTransport.newStream()`, enters the while-loop, gets null transport because it's still CONNECTING, not yet enters the synchronized block.
2. T2: transport CONNECTING->READY, calls `updatePicker()` with the same picker
3. T1:  enters the synchronized block, finds `picker == lastPicker`, even though `lastPicker` was rewritten after `newStream()` started.
4. Outcome: the stream will be buffered, even though the Subchannel is already READY.

Of course the pick-first LB could work this around by excluding CONNECTING Subchannels, thus returning `withNoResult()` for CONNECTING state, forcing a different picker to be created for the `CONNECTING->READY` transition, but the version number actually **fixes an inconsistency in delayed transport**: `reprocess()` does not ignore identical pickers, but the while-loop in `newStream()`, which is meant to solve the race between `reprocess()` and `newStream(), did ignore identical pickers.